src/ast.rs

```diff
@@ -366,28 +375,42 @@ impl PartialOrd for RawBinaryOperator {
     }
 }
 
+/// This serves a similar purpose to ParenthesizedExpr.
+///
+/// see <https://google.github.io/xls/dslx_reference/#block-expressions>
+pub struct BlockExpr(pub Expr);
+
 /// *Part* of a let binding: the variable declaration and value it is bound to. What is missing
 /// is the expression in which the bound name is in-scope.
```

Comment:
OOC, in your picture of the AST, do elements have a "parent" reference? My memory is rusty, as you might imagine, no run intended.